src/System.Web.HttpUtility/tests/HttpUtility/HttpUtilityTest.cs

```diff
@@ -753,5 +753,16 @@ public void UrlPathEncode(string decoded, string encoded)
         {
             Assert.Equal(encoded, HttpUtility.UrlPathEncode(decoded));
         }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("name=foo&desc=foo")]
+        [InlineData("type=foo&type=foo")]
```

Comment:
It would be good to have a few more test cases here. For example, how are empty values (e.g. `name=&desc=foo`) handled? Does that behavior change with the fix?

Other than that, I think the change looks good. Thanks for fixing this.

Reply:
Good idea. In that case I would recommend `name=&name=foo` as well.
.NET Framework code had special entries for `name&another`, which is not part of this change, but I bet those things don't work in compatible way either ...

Reply:
> .NET Framework code had special entries for `name&another`, which is not part of this change, but I bet those things don't work in compatible way either ...

No, this isn't currently handled, and my PR doesn't change that.